### ISSUE TITLE
Error retrieving JSON Logger Config with External Destination (Anypoint MQ)

### ISSUE THREAD

**Issue by lealpm** [anypointcloud/json-logger#4]
We have a commons plugin containing a few global configurations like Salesforce, Database and Json logger. Importing this dependency and trying to reuse this configuration everything works fine for Salesforce and Database but we are facing an error with JSON Logger.

The error is the following: _Cannot invoke "org.mule.extension.jsonlogger.internal.destinations.Destination.getSelectedDestinationType()" because the return value of "java.util.HashMap.get(Object)" is null_

Important to notice that this only occurs when the JSON Logger configuration is configured with an External Destination (Anypoint MQ).

**Comment by mtransier** [comment]
Hi @lealpm, would it be possible for you to provide a stripped down version of your plugin and app so that we can reproduce it easily?

**Comment by lealpm** [comment]
Yes, sure but first it would be great if you can have a look on the dependencies of the original project. When I try to deploy an app using the latest JSON Logger version, configured with an Anypoint MQ as an External Destination, I get this error:

_**+ Failed to deploy artifact 'jsonloggertestapp',                               +
+ org.mule.runtime.deployment.model.api.DeploymentInitException:               +
+ ClassNotFoundException: com.google.gson.reflect.TypeToken**_       

We were able to "fix" it by adding and excluding these dependencies:

![image](https://github.com/user-attachments/assets/89b411f3-6024-492b-a4f0-155d6452685f)

Can you check this fix and evaluate if it's correct or if there is a more proper way of fixing it?

**Comment by mtransier** [comment]
Hi @lealpm, please have a look at the current SNAPSHOT version in the main branch? Unfortunately, it is not available from the maven repository, so you will have to build it yourself from source. This issue is already fixed there, the logger is working with Anypoint MQ.

**Comment by mtransier** [comment]
Hi @lealpm, can you try again with the latest version in the main branch?

**Comment by mtransier** [comment]
Solved with release 2.2.2 (commons-lang was missing). Feel free to create a new issue if it persists. And in that case, please also provide an example to reproduce the issue if possible.
